vllm/entrypoints/pooling/embed/serving.py

```diff
@@ -536,8 +536,8 @@ async def _collect_batch(
                     # Non-chunked result - extract prompt_idx from request_id
                     parts = result.request_id.split("-")
                     try:
-                        # Last part should be prompt index
-                        prompt_idx = int(parts[-1])
+                        # Second-to-last part should be prompt index
+                        prompt_idx = int(parts[-2])
```

Comment:
Yuck, the way the request id is abused in this file is pretty bad. What if an external request id is sent in that happens to contains `-chunk-` or whatever.

(not related to your changes, just hadn't seen this code before!)

Reply:
Agree!